python/hail/matrixtable.py

```diff
@@ -1663,20 +1663,24 @@ def group_rows_by(self, *exprs, **named_exprs):
 
         Notes
         -----
-        The `key_expr` argument can either be a string referring to a row-indexed field
-        of the dataset, or an expression that will become the new row key.
+        All complex expressions (expressions that aren't simply row fields) must
+        be passed in as a keyword argument and are assigned to a field.
```

Comment:
"must be passed as named expressions" (that's a concept we use elsewhere I think)